plugins/modules/s3_metrics_configuration.py

```diff
@@ -0,0 +1,221 @@
+    state: present
+
+- name: Put a metrics configuration that enables metrics for objects with specific tag
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: Assets
+    filter_tag:
+      kind: asset
+    state: present
+
+- name: Put a metrics configuration that enables metrics for objects that start with a particular prefix and have specific tags applied
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: ImportantBlueDocuments
+    filter_prefix: documents
+    filter_tags:
+      priority: high
+      class: blue
+    state: present
+
+- name: Delete metrics configuration
+  community.aws.s3_metrics_configuration:
+    bucket_name: my-bucket
+    id: EntireBucket
+    state: absent
+
+'''
+
+try:
+    import boto3
+    import botocore
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule, is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry, ansible_dict_to_boto3_tag_list
+
+
+def _create_metrics_configuration(mc_id, filter_prefix, filter_tags):
+    payload = {
+        'Id': mc_id
+    }
+    # Just a filter_prefix or just a single tag filter is a special case
+    if filter_prefix and not filter_tags:
+        payload['Filter'] = {
+            'Prefix': filter_prefix
+        }
+    elif not filter_prefix and len(filter_tags) == 1:
+        payload['Filter'] = {
+            'Tag': ansible_dict_to_boto3_tag_list(filter_tags)[0]
+        }
+    # Otherwise we need to use 'And'
+    elif filter_tags:
+        payload['Filter'] = {
+            'And': {
+                'Tags': ansible_dict_to_boto3_tag_list(filter_tags)
+            }
+        }
+        if filter_prefix:
+            payload['Filter']['And']['Prefix'] = filter_prefix
+
+    return payload
+
+
+def create_or_update_metrics_configuration(client, module):
+    bucket_name = module.params.get('bucket_name')
+    mc_id = module.params.get('id')
+    filter_prefix = module.params.get('filter_prefix')
+    filter_tags = module.params.get('filter_tags')
+
+    try:
+        response = client.get_bucket_metrics_configuration(aws_retry=True, Bucket=bucket_name, Id=mc_id)
+        metrics_configuration = response['MetricsConfiguration']
+    except is_boto3_error_code('NoSuchConfiguration'):
+        metrics_configuration = None
+    except (BotoCoreError, ClientError) as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to get bucket metrics configuration")
+
+    new_configuration = _create_metrics_configuration(mc_id, filter_prefix, filter_tags)
+
+    if metrics_configuration:
+        if metrics_configuration == new_configuration:
+            module.exit_json(changed=False)
+        
+        if module.check_mode:
+            module.exit_json(changed=True)
```

Comment:
```suggestion
    if module.check_mode:
        module.exit_json(changed=True)
```

The current indentation means that even in check_mode, if no configuration exists, then we'll add it.

Reply:
Nice spot 👍